File: Dataset/PalindromeChecker.py
import aspectlib
@aspectlib.Aspect
def logging_aspect(cutpoint):
    print(f"Method called: {cutpoint.__name__}")
    yield aspectlib.Proceed
@logging_aspect
def is_palindrome(number):
    original = number
    reversed_number = 0
    while number != 0:
        digit = number % 10
        reversed_number = reversed_number * 10 + digit
        number //= 10
    return original == reversed_number
if __name__ == "__main__":
    number = 121  # Example number
    result = is_palindrome(number)
    print(f"Is {number} a palindrome? {result}")

File: Dataset/PalindromeChecker.cs
using PostSharp.Aspects;
using System;
[Serializable]
public class LoggingAspect : OnMethodBoundaryAspect {
    public override void OnEntry(MethodExecutionArgs args) {
        Console.WriteLine("Method called: " + args.Method.Name);
    }
}
public class PalindromeChecker {
    [LoggingAspect]
    public bool IsPalindrome(int number) {
        int original = number;
        int reversed = 0;
        while (number != 0) {
            int digit = number % 10;
            reversed = reversed * 10 + digit;
            number /= 10;
        }
        return original == reversed;
    }
    static void Main(string[] args) {
        PalindromeChecker checker = new PalindromeChecker();
        int number = 121;
        bool result = checker.IsPalindrome(number);
        Console.WriteLine("Is " + number + " a palindrome? " + result);
    }
}
